From b3c5ce68f23d78b39284f0fc43c4313a075310f5 Mon Sep 17 00:00:00 2001
From: Josh ben Jore <jbenjore@whitepages.com>
Date: Sat, 9 May 2009 22:09:27 -0700
Subject: [PATCH] Use -Dman3ext

---
 t/inc/conf.pl |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/t/inc/conf.pl b/t/inc/conf.pl
index c6248be..066aa8e 100755
--- a/t/inc/conf.pl
+++ b/t/inc/conf.pl
@@ -59,7 +59,7 @@ BEGIN {
     $CONTENTS = {
         xs  => 
             [       # ones we need
-                [   $man3 . q[/Foo::Bar.3pm],
+                [   $man3 . q[/Foo::Bar.] . $Config{'man3ext'},
                     $arch . q[/Foo/Bar.pm],
                     $arch . q[/auto/Foo/Bar/Bar.bs],
                     $arch . q[/auto/Foo/Bar/Bar.so],
@@ -72,7 +72,7 @@ BEGIN {
             ],
         noxs => 
             [       # ones we need
-                [   $man3 . q[/Foo::Bar.3pm],
+                [   $man3 . q[/Foo::Bar.] . $Config{'man3ext'},
                     $pref . q[/Foo/Bar.pm],
                     $bin  . q[/foobar.pl],
                 ],
-- 
1.6.0.4

