From 48bef09c5f349df6371b9b3be0ef919ff0da910e Mon Sep 17 00:00:00 2001
From: Josh ben Jore <jbenjore@whitepages.com>
Date: Sat, 9 May 2009 22:48:31 -0700
Subject: [PATCH] Core modules depend only on perl

---
 lib/CPANPLUS/Dist/Deb.pm |   55 ++++++++++++++++++++++++++--------------------
 1 files changed, 31 insertions(+), 24 deletions(-)

diff --git a/lib/CPANPLUS/Dist/Deb.pm b/lib/CPANPLUS/Dist/Deb.pm
index 588a4ac..74d8fb5 100755
--- a/lib/CPANPLUS/Dist/Deb.pm
+++ b/lib/CPANPLUS/Dist/Deb.pm
@@ -566,30 +566,37 @@ sub prepare {
         ### >= VERSION if the prereqs were stated with requiring a certain
         ### version.. otherwise we leave it empty
         my %seen;
-        my $prereqs         = join ', ', map {
-                                ### do we need a specific version?
-                                my $ver = $_->[1] 
-                                            ? ' (>= ' . $_->[1] . ')' 
-                                            : '';
-                                
-                                ### standard lib
-                                my $str = DEB_PACKAGE_NAME->($_->[0]) . $ver;
-
-                                ### our lib, if it has a prefix
-                                if( $prefix ) {
-                                    $str .= ' | ' . DEB_PACKAGE_NAME->(
-                                            $_->[0], $prefix) . $ver;
-                                }
-                                
-                                $str;
-                            } grep {
-                                ### shouldn't be a core module
-                                ### and we shouldn't list the same
-                                ### prereq twice. Note that 2 modules
-                                ### may be in 1 package
-                                !$_->[0]->module_is_supplied_with_perl_core and
-                                !$seen{ DEB_PACKAGE_NAME->( $_->[0] ) }++
-                            } @depends;
+        my $prereqs;
+        if ( $self->module_is_supplied_with_perl_core ) {
+            $prereqs = DEB_THIS_PERL_DEPENDS->();
+        } else {
+            $prereqs =
+                join ', ',
+                map {
+                    ### do we need a specific version?
+                    my $ver = $_->[1]
+                                ? ' (>= ' . $_->[1] . ')'
+                                : '';
+
+                    ### standard lib
+                    my $str = DEB_PACKAGE_NAME->($_->[0]) . $ver;
+
+                    ### our lib, if it has a prefix
+                    if( $prefix ) {
+                        $str .= ' | ' . DEB_PACKAGE_NAME->(
+                                $_->[0], $prefix) . $ver;
+                    }
+
+                    $str;
+                } grep {
+                    ### shouldn't be a core module
+                    ### and we shouldn't list the same
+                    ### prereq twice. Note that 2 modules
+                    ### may be in 1 package
+                    !$_->[0]->module_is_supplied_with_perl_core and
+                    !$seen{ DEB_PACKAGE_NAME->( $_->[0] ) }++
+                } @depends;
+        }
 
         ### always put debhelper in build-depends ###
         my $build_depends   =  $debhelper;
-- 
1.6.0.4

