From 806c07e8c32ed035ade92c5e905372be455d9131 Mon Sep 17 00:00:00 2001
From: Josh ben Jore <jbenjore@whitepages.com>
Date: Sat, 9 May 2009 22:48:55 -0700
Subject: [PATCH] Rename %seen to %seen_prereq

---
 lib/CPANPLUS/Dist/Deb.pm |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/CPANPLUS/Dist/Deb.pm b/lib/CPANPLUS/Dist/Deb.pm
index 74d8fb5..93399b5 100755
--- a/lib/CPANPLUS/Dist/Deb.pm
+++ b/lib/CPANPLUS/Dist/Deb.pm
@@ -565,7 +565,7 @@ sub prepare {
         ### '${prefix}libfoo-perl' if we do have a prefix. We only add the
         ### >= VERSION if the prereqs were stated with requiring a certain
         ### version.. otherwise we leave it empty
-        my %seen;
+        my %seen_prereq;
         my $prereqs;
         if ( $self->module_is_supplied_with_perl_core ) {
             $prereqs = DEB_THIS_PERL_DEPENDS->();
@@ -594,7 +594,7 @@ sub prepare {
                     ### prereq twice. Note that 2 modules
                     ### may be in 1 package
                     !$_->[0]->module_is_supplied_with_perl_core and
-                    !$seen{ DEB_PACKAGE_NAME->( $_->[0] ) }++
+                    !$seen_prereq{ DEB_PACKAGE_NAME->( $_->[0] ) }++
                 } @depends;
         }
 
-- 
1.6.0.4

