From 6789ba5a7de2c0fc329611c972d666ed0389368c Mon Sep 17 00:00:00 2001
From: Josh ben Jore <jbenjore@whitepages.com>
Date: Sun, 10 May 2009 00:22:06 -0700
Subject: [PATCH] Remove .deb work product

---
 t/02_debs.t |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/t/02_debs.t b/t/02_debs.t
index eb4c265..d00617d 100755
--- a/t/02_debs.t
+++ b/t/02_debs.t
@@ -182,4 +182,8 @@ if( @ARGV ) {
             }
         }    
     }
+
+    # Moar st00pit vms
+    1 while unlink $deb;
+    ok( !-e $deb, "'$deb' got deleted" );
 }
-- 
1.6.0.4

