From f9295e82c72783ba7c94dc710cb8e7badb7b8277 Mon Sep 17 00:00:00 2001
From: Josh ben Jore <jbenjore@whitepages.com>
Date: Sun, 10 May 2009 00:32:36 -0700
Subject: [PATCH] Test files in a predictable order for easier test comparison

---
 t/02_debs.t |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/t/02_debs.t b/t/02_debs.t
index b45c39b..77933de 100755
--- a/t/02_debs.t
+++ b/t/02_debs.t
@@ -106,11 +106,11 @@ for my $test ( @TESTS ) {
                 
                 ok( $out,                   "   Deb --contents retrieved" );
 
-                for my $entry ( @$need ) {
+                for my $entry ( sort @$need ) {
                     my $re = qr/.$entry$/m;
                     like( $out, $re,        "       Contains $entry" );
                 }
-                for my $entry ( @$omit ) {
+                for my $entry ( sort @$omit ) {
                     my $re = qr/$entry$/m;
                     unlike( $out, $re,      "       Doesn't contain $entry" );
                 }
@@ -131,14 +131,14 @@ for my $test ( @TESTS ) {
                 ok( $out,                   "       Deb files retrieved" );
             
                 ### check out if all got installed ok
-                for my $entry ( @$need ) {
+                for my $entry ( sort @$need ) {
                     ok( -e $entry,          "       File $entry installed" );                
                 }
 
                 ok( $dist->uninstall,       "   Dist uninstalled" );
                 
                 ### check out if all got removed ok
-                for my $entry ( @$need ) {
+                for my $entry ( sort @$need ) {
                     ok( !-e $entry,         "       File $entry uninstalled");                
                 }
                 
@@ -148,7 +148,7 @@ for my $test ( @TESTS ) {
                 ok( $files,                 "   Files for this package" );
                 is( scalar(@$files), 5,     "       All Files found" );
                 
-                for ( @$files ) {
+                for ( sort @$files ) {
                     ok( -e $_,              "       File '$_' exists" );
                     1 while unlink $_;
                     ok( !-e $_,             "       File '$_' removed" );
-- 
1.6.0.4

