From ea3c3e16f2a266ff31fedafc40a4267d37336670 Mon Sep 17 00:00:00 2001
From: Josh ben Jore <jbenjore@whitepages.com>
Date: Sun, 10 May 2009 00:35:47 -0700
Subject: [PATCH] Skip fatal tests

---
 t/02_debs.t |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/t/02_debs.t b/t/02_debs.t
index 77933de..e54b4f3 100755
--- a/t/02_debs.t
+++ b/t/02_debs.t
@@ -146,6 +146,10 @@ for my $test ( @TESTS ) {
            
             {   my $files = $dist->status->files;
                 ok( $files,                 "   Files for this package" );
+
+      SKIP: {
+            skip( 'Missing files', 5 ) if ! $files;
+
                 is( scalar(@$files), 5,     "       All Files found" );
                 
                 for ( sort @$files ) {
@@ -154,6 +158,7 @@ for my $test ( @TESTS ) {
                     ok( !-e $_,             "       File '$_' removed" );
                 }
             }
+    }
             
             ### check out the naming
             like( $deb, qr/$DEBMOD/,        "   Deb is called '$DEBMOD'" );
-- 
1.6.0.4

